### PR TITLE
Apt improvements

### PR DESCRIPTION
**Pull request by feliwir**

### CODE REVIEW

File: src/OpenSage.Game/Gui/Apt/ActionScript/Library/Builtin.cs [hunk 1]
```diff
@@ -125,9 +155,23 @@ private static void Stop(ObjectContext ctx)
             }
         }
 
-        private static void SetInterval(ObjectContext ctx,Value[] args)
+        private static void SetInterval(ActionContext actionContext, ObjectContext ctx, Value[] args)
+        {
+            var vm = actionContext.Apt.AVM;
+            var name = actionContext.Stack.Pop().ToString();
+
+            vm.CreateInterval(name, args[1].ToInteger(), args[0].ToFunction(), ctx, new Value[0]);
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/ActionScript/Library/Builtin.cs]: Use `Array.Empty<Value>()`.

File: src/OpenSage.Game/Gui/Apt/ActionScript/Value.cs [hunk 2]
```diff
@@ -156,6 +162,10 @@ public ObjectContext ToObject()
 
         public int ToInteger()
         {
+            if (Type == ValueType.Undefined ||
+                Type == ValueType.String)
+                return 0;
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/ActionScript/Value.cs]: Add braces here.

File: src/OpenSage.Game/Gui/Apt/AptCallbacksAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace OpenSage.Gui.Apt
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public sealed class AptCallbacksAttribute : Attribute
+    {
+        public SageGame[] Games { get; private set; }
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/AptCallbacksAttribute.cs]: No need for the setter.

File: src/OpenSage.Game/Gui/Apt/AptContext.cs [hunk 4]
```diff
@@ -8,13 +8,11 @@ namespace OpenSage.Gui.Apt
...
-        private ImageMap _imageMap;
-        private string _movieName;
-        private VM _avm;
-    
-        public VM AVM => _avm;
-        public ContentManager ContentManager => _contentManager;
+        private readonly ImageMap _imageMap;
+        private readonly string _movieName;
+
+        public VM AVM { get; }
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/AptContext.cs]: This should be called `Avm`.

File: src/OpenSage.Game/Gui/Apt/AptInputMessageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using OpenSage.Input;
+
+namespace OpenSage.Gui.Apt
+{
+    class AptInputMessageHandler : InputMessageHandler
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/AptInputMessageHandler.cs]: Please add `internal sealed` here.

File: src/OpenSage.Game/Gui/Apt/DisplayItem.cs [hunk 6]
```diff
@@ -26,6 +26,13 @@ public ItemTransform(in ColorRgbaF color, in Matrix3x2 rotation, in Vector2 tran
                                      a.GeometryRotation * b.GeometryRotation,
                                      a.GeometryTranslation + b.GeometryTranslation);
         }
+
+        public ItemTransform WithColorTransform(ColorRgbaF color)
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/DisplayItem.cs]: Add `in` modifier.

File: src/OpenSage.Game/Gui/Apt/RenderItem.cs [hunk 7]
```diff
@@ -1,29 +1,26 @@
...
     {
-        private SpriteItem _parent;
-        private Character _character;
-        private AptContext _context;
-        private ObjectContext _scriptObject;
-
-        public SpriteItem Parent => _parent;
-        public Character Character => _character;
-        public AptContext Context => _context;
+        public SpriteItem Parent { get; private set; }
```
Comment (tgjones) [inline src/OpenSage.Game/Gui/Apt/RenderItem.cs]: I think all these private setters can be removed entirely.

Comment (feliwir) [reply src/OpenSage.Game/Gui/Apt/RenderItem.cs]: Doesn't work for me when removing them. I guess i can only set them from the constructor without the `private set`

**Pull request merged**